ts/sveltelib/iterate-actions.ts

```diff
@@ -0,0 +1,37 @@
+// Copyright: Ankitects Pty Ltd and contributors
+// License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+
+import { truthy } from "../lib/functional";
+
+interface ActionReturn<P> {
+    destroy?(): void;
+    update?(params: P): void;
+}
+
+type Action<E extends HTMLElement, P> = (
+    element: E,
+    params: P,
+) => ActionReturn<P> | void;
+
+function iterateActions<E extends HTMLElement, P>(
```

Comment:
docstring would be nice

Reply:
Thanks, that makes it clearer. The current name makes me think that the actions will be iterated when that function is called, instead of when its return value is used. Would something like `actionList` or `groupedActions` make sense instead?